hudi-client/hudi-spark-client/src/test/java/org/apache/hudi/client/functional/TestHoodieBackedMetadata.java

```diff
@@ -1261,17 +1532,190 @@ private void validateMetadata(SparkRDDWriteClient testClient) throws IOException
     LOG.info("Validation time=" + timer.endTimer());
   }
 
+  /**
+   * Validate the metadata tables contents to ensure it matches what is on the file system.
+   */
+  private void validateMetadata(HoodieTestTable testTable, HoodieWriteConfig config, HoodieEngineContext hoodieEngineContext) throws IOException {
+    validateMetadata(testTable, config, hoodieEngineContext, Collections.emptyList());
+  }
+
+  /**
+   * Validate the metadata tables contents to ensure it matches what is on the file system.
+   */
+  private void validateMetadata(HoodieTestTable testTable, HoodieWriteConfig config, HoodieEngineContext hoodieEngineContext,
```

Comment:
Note to Reviewer: This is an almost replica of existing ValidateMetadata except that this uses HoodieTestTable as source table for validation. As mentioned in the description, will be adding more tests and eventually will remove direct SparkRDDClient based tests.